src/mrack/providers/aws.py

```diff
@@ -146,11 +140,11 @@ async def wait_till_provisioned(self, aws_id):
         instance = self.ec2.Instance(aws_id)
         instance.wait_until_running()
         response = self.client.describe_instances(InstanceIds=[aws_id])
-        assert len(response["Reservations"]) == 1
-        assert len(response["Reservations"][0]["Instances"]) == 1
-        result = response["Reservations"][0]["Instances"][0]
-        # returns dict with aws instance information
-        return result
+
+        try:  # returns dict with aws instance information
+            return response["Reservations"][0]["Instances"][0]
+        except (KeyError, IndexError):
+            raise (ProvisioningError)
```

Comment:
```python
raise ProvisioningError("Missing data provisioned instance.")
```
_That is just a suggestion, feel free to improve the message._

Reply:
Will do :)